src/WebJobs.Extensions.Http/ClaimsIdentityBindingProvider.cs

```diff
@@ -0,0 +1,206 @@
+                }
+
+                var bindingData = context.BindingData;
+                var claimsPrincipal = ClaimsPrincipalHelper.FromBindingData(bindingData);
+
+                if (_bindingType == typeof(ClaimsIdentity))
+                {
+                    if (claimsPrincipal.Identities.Count() == 1)
+                    {
+                        return Task.FromResult<IValueProvider>(new SingleClaimsIdentityValueProvider(claimsPrincipal.Identities.First()));
+                    }
+
+                    var identityProvider = GetIdentityProvider(bindingData);
+                    var identity = GetPrimaryIdentity(claimsPrincipal, identityProvider);
+                    return Task.FromResult<IValueProvider>(new SingleClaimsIdentityValueProvider(identity));
+                }
+                else if (_bindingType == typeof(IEnumerable<ClaimsIdentity>))
+                {
+                    return Task.FromResult<IValueProvider>(new ManyClaimsIdentityValueProvider(claimsPrincipal.Identities));
+                }
+                else
+                {
+                    throw new InvalidOperationException($"The type {_bindingType} is not supported");
+                }
+            }
+
+            public Task<IValueProvider> BindAsync(object value, ValueBindingContext context)
+            {
+                throw new NotImplementedException("This method doesn't have the necessary binding data.");
+            }
+
+            public ParameterDescriptor ToParameterDescriptor()
+            {
+                return new ParameterDescriptor
+                {
+                    Name = _parameter.Name,
+                    DisplayHints = new ParameterDisplayHints
+                    {
+                        Description = "ClaimsIdentity"
+                    }
+                };
+            }
+
+            private static string GetIdentityProvider(IReadOnlyDictionary<string, object> bindingData)
+            {
+                if (bindingData.TryGetValue(HttpTriggerAttributeBindingProvider.HttpHeadersKey, out object value))
+                {
+                    IDictionary<string, string> headers = (IDictionary<string, string>)value;
+                    if (headers != null)
+                    {
+                        headers.TryGetValue(HttpConstants.AntaresEasyAuthProviderHeaderName, out string identityProvider);
+                        return identityProvider;
+                    }
+                }
+                return null;
+            }
+
+            private static ClaimsIdentity GetPrimaryIdentity(ClaimsPrincipal claimsPrincipal, string identityProvider = null)
+            {
+                // if the principal contains a key based identity that will take precedence
+                // over any other identity
+                var identity = claimsPrincipal.Identities.LastOrDefault(p => p.IsAuthenticated && string.Equals(p.AuthenticationType, "key", StringComparison.OrdinalIgnoreCase));
```

Comment:
As Chris mentioned earlier, I don't think the "key" identity is present here, even when using key based authentication for an HTTP function.

Reply:
Right - that was additional prototype work that I had done in a separate PR to add that identity, to ensure a consistent experience.